datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/DatalakeHorizontalScaleActions.java

```diff
@@ -0,0 +1,275 @@
+                        updateRequest.getDesiredCount());
+                updateRequest.setStackId(sdxCluster.getStackId());
+                updateRequest.setAdjustmentType(AdjustmentType.EXACT);
+                try {
+                    flowId = ThreadBasedUserCrnProvider.doAsInternalActor(
+                            regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                            () -> stackV4Endpoint.putScaling(0L, sdxCluster.getClusterName(), updateRequest, sdxCluster.getAccountId())
+                    ).getPollableId();
+                } catch (NotFoundException e) {
+                    LOGGER.info("Cannot find stack on cloudbreak side {}", sdxCluster.getClusterName());
+                } catch (WebApplicationException e) {
+                    String errorMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
+                    LOGGER.info("Cannot horizontal scale stack {} from cloudbreak: {}", sdxCluster.getStackId(), errorMessage, e);
+                    throw new RuntimeException(errorMessage);
+                } catch (ProcessingException e) {
+                    throw new RuntimeException(e.getMessage());
+                }
+                LOGGER.info("Put scaling flow triggered for datalake horziontal scale. FlowId: {}", flowId);
+                var selector = DATALAKE_HORIZONTAL_SCALE_IN_PROGRESS_HANDLER.selector();
+                var event = new DatalakeHorizontalScaleStartRequest(selector,
+                        payload.getResourceId(),
+                        sdxCluster.getName(),
+                        sdxCluster.getResourceCrn(),
+                        flowId,
+                        payload.getUserId(),
+                        payload.getStackScaleRequest().getGroup(),
+                        payload.getStackScaleRequest().getDesiredCount());
+                sendEvent(context, selector, event);
+            }
+
+            @Override
+            protected Object getFailurePayload(DatalakeHorizontalScaleStartEvent payload, Optional<SdxContext> flowContext, Exception ex) {
+                LOGGER.error("Datalake horizontal scale failed to create new instance.", ex);
+                var selector = DATALAKE_HORIZONTAL_SCALE_FAILED_EVENT.selector();
+                var context = flowContext.orElseGet(() -> new SdxContext(null, payload.getResourceId(), payload.getUserId()));
+                return new DatalakeHorizontalScaleFailedEvent(
+                        selector,
+                        payload.getResourceId(),
+                        payload.getSdxName(),
+                        payload.getResourceCrn(),
+                        payload.getUserId(),
+                        payload.getStackScaleRequest().getGroup(),
+                        payload.getStackScaleRequest().getDesiredCount(),
+                        ex);
+            }
+        };
+    }
+
+    @Bean(name = "DATALAKE_HORIZONTAL_SCALE_CM_RESTART_STATE")
```

Comment:
In this state we will be restarting the services, not CM, right?
In that case, the appropriate name might be  DATALAKE_HORIZONTAL_SCALE_SERVICES_RESTART_STATE. What do you think?